ui/lib/kv/addon/components/page/secret/metadata-edit.hbs

```diff
@@ -1,6 +1,60 @@
-<KvPageHeader @breadcrumbs={{@breadcrumbs}} @model={{@model}} @pageTitle="Edit Secret Metadata" />
+<KvPageHeader @breadcrumbs={{@breadcrumbs}} @pageTitle="Edit Secret Metadata" />
 
 <hr class="is-marginless has-background-gray-200" />
-<p class="has-top-margin-m">
+<p class="has-top-margin-m has-bottom-margin-m">
   The options below are all version-agnostic; they apply to all versions of this secret.
-</p>
\ No newline at end of file
+</p>
+{{#if @metadata.canUpdateMetadata}}
+  <form {{on "submit" (perform this.save)}}>
+    <div class="box is-sideless is-fullwidth is-marginless">
+      <NamespaceReminder @mode="update" @noun="KV secret metadata" />
+      <MessageError @errorMessage={{this.errorBanner}} class="has-top-margin-s" />
+      {{#each @metadata.formFields as |attr|}}
+        {{#if (eq attr.name "customMetadata")}}
+          <FormField @attr={{attr}} @model={{@metadata}} @modelValidations={{this.modelValidations}} />
+          <label class="title has-top-padding-m is-4">
+            Additional options
+          </label>
+        {{else if (not-eq attr.name "customMetadata")}}
+          <FormField @attr={{attr}} @model={{@metadata}} @modelValidations={{this.modelValidations}} />
+        {{/if}}
+      {{/each}}
+    </div>
+    <div class="field is-grouped is-grouped-split is-fullwidth box is-bottomless">
+      <div class="has-top-padding-s">
+        <button
+          type="submit"
+          class="button is-primary {{if this.save.isRunning 'is-loading'}}"
+          disabled={{this.save.isRunning}}
+          data-test-kv-metadata-save
+        >
+          Update
+        </button>
+        <button
+          type="button"
+          class="button has-left-margin-s"
+          disabled={{this.save.isRunning}}
+          {{on "click" this.cancel}}
+          data-test-kv-metadata-cancel
+        >
+          Cancel
+        </button>
+        {{#if this.invalidFormAlert}}
+          <div class="control">
+            <AlertInline @type="danger" @paddingTop={{true}} @message={{this.invalidFormAlert}} @mimicRefresh={{true}} />
+          </div>
+        {{/if}}
+      </div>
+    </div>
+  </form>
+{{else}}
+  <EmptyState
+    @title="You do not have permissions to edit metadata"
+    @message="Ask your administrator if you think you should have access."
+  >
+    <LinkTo @route="secret.metadata.index" @model={{@metadata.path}} class="link">
```

Comment:
I've noticed that I don't think we need this `link` class anymore (at least in the version dropdown menu I didn't..)

Reply:
Makes sense, let me remove (I need to retrigger this stalled out setup test thing anyways).